exchanges/ftx/ftx.go

```diff
@@ -0,0 +1,785 @@
+	deleteMyQuote            = "/options/quotes/%s"
+	acceptQuote              = "/options/quotes/%s/accept"
+	getOptionsInfo           = "/options/account_info"
+	getOptionsPositions      = "/options/positions"
+	getPublicOptionsTrades   = "/options/trades"
+	getOptionsFills          = "/options/fills"
+	ftxRateInterval          = time.Minute
+	ftxRequestRate           = 180
+
+	// Other Consts
+	stopOrderType         = "stop"
+	trailingStopOrderType = "trailingStop"
+	takeProfitOrderType   = "takeProfit"
+	buy                   = "buy"
+	sell                  = "sell"
+	newStatus             = "new"
+	openStatus            = "open"
+	closedStatus          = "closed"
+	limitOrder            = "limit"
+	marketOrder           = "market"
+	defaultTime           = "0001-01-01 00:00:00 +0000 UTC"
+	spotString            = "spot"
+	futuresString         = "future"
+)
+
+// Start implementing public and private exchange API funcs below
+
+// GetMarkets gets market data
+func (f *FTX) GetMarkets() (Markets, error) {
+	var resp Markets
+	return resp, f.SendHTTPRequest(ftxAPIURL+getMarkets, &resp)
+}
+
+// GetMarket gets market data for a provided asset type
+func (f *FTX) GetMarket(marketName string) (Market, error) {
+	var market Market
+	return market, f.SendHTTPRequest(fmt.Sprintf(ftxAPIURL+getMarket, marketName),
+		&market)
+}
+
+// GetOrderbook gets orderbook for a given market with a given depth (default depth 20)
+func (f *FTX) GetOrderbook(marketName string, depth int64) (OrderbookData, error) {
+	strDepth := strconv.FormatInt(depth, 10)
+	var resp OrderbookData
+	var tempOB TempOrderbook
+	err := f.SendHTTPRequest(fmt.Sprintf(ftxAPIURL+getOrderbook, marketName, strDepth), &tempOB)
+	if err != nil {
+		return resp, err
+	}
+	resp.MarketName = marketName
+	for x := range tempOB.Result.Asks {
+		resp.Asks = append(resp.Asks, OData{Price: tempOB.Result.Asks[x][0],
+			Size: tempOB.Result.Bids[x][1],
+		})
+	}
+	for y := range tempOB.Result.Bids {
+		resp.Bids = append(resp.Bids, OData{Price: tempOB.Result.Bids[y][0],
+			Size: tempOB.Result.Bids[y][1],
+		})
+	}
+	return resp, nil
+}
+
+// GetTrades gets trades based on the conditions specified
+func (f *FTX) GetTrades(marketName, startTime, endTime string, limit int64) (Trades, error) {
+	strLimit := strconv.FormatInt(limit, 10)
+	params := url.Values{}
+	params.Set("limit", strLimit)
+	var sTime, eTime int64
+	var err error
+	var resp Trades
+	if startTime != "" {
+		sTime, err = strconv.ParseInt(startTime, 10, 64)
+		if err != nil {
+			return resp, err
+		}
+		params.Set("start_time", startTime)
+	}
+	if endTime != "" {
+		eTime, err = strconv.ParseInt(endTime, 10, 64)
+		if err != nil {
+			return resp, err
+		}
+		params.Set("end_time", endTime)
+	}
+	if startTime != "" && endTime != "" {
+		if sTime > eTime {
+			return resp, errors.New("startTime cannot be bigger than endTime")
+		}
+	}
+	return resp, f.SendHTTPRequest((fmt.Sprintf(ftxAPIURL+getTrades, marketName) + params.Encode()),
```

Comment:
unneeded parentheses (although you should still update this following @gloriousCode's comment from above